rs/ethereum/cketh/minter/src/eth_rpc_client/mod.rs

```diff
@@ -670,6 +678,48 @@ impl Reduce for MultiCallResults<FeeHistory> {
     }
 }
 
+impl Reduce for EvmMultiRpcResult<Option<EvmTransactionReceipt>> {
+    type Item = Option<TransactionReceipt>;
+
+    fn reduce(self) -> ReducedResult<Self::Item> {
+        fn map_transaction_receipt(
+            receipt: Option<EvmTransactionReceipt>,
+        ) -> Result<Option<TransactionReceipt>, String> {
+            receipt
+                .map(|evm_receipt| {
+                    Ok(TransactionReceipt {
+                        block_hash: Hash::from_str(&evm_receipt.block_hash)?,
+                        block_number: BlockNumber::try_from(evm_receipt.block_number)?,
+                        effective_gas_price: WeiPerGas::try_from(evm_receipt.effective_gas_price)?,
+                        gas_used: GasAmount::try_from(evm_receipt.gas_used)?,
+                        status: TransactionStatus::try_from(
+                            evm_receipt
+                                .status
+                                .0
+                                .to_u8()
+                                .ok_or("invalid transaction status")?,
+                        )?,
+                        transaction_hash: Hash::from_str(&evm_receipt.transaction_hash)?,
+                    })
+                })
+                .transpose()
```

Comment:
Why `transpose`?

Reply:
This allows to transform an `Option<Result>` into a `Result<Option>`, see the [docs](https://doc.rust-lang.org/std/option/enum.Option.html#method.transpose)

Reply:
Please don't hesitate to let me know if this didn't answer your question!